### PR TITLE
Feature/fga/dial pad improvements

### PR DESCRIPTION
**Pull request by ganfra**
This PR handles: 
Enabling cursor, so you can add/remove digits everywhere.
Fixing the formatting as you type 
Adding paste number when clicking on ok button if EditText is empty (mimics Android DialPad)

### CODE REVIEW

File: vector/src/main/java/im/vector/app/features/call/dialpad/DialPadFragment.kt [hunk 1]
```diff
@@ -145,40 +158,40 @@ class DialPadFragment : Fragment() {
...
-    private fun append(c: Char) {
-        callback?.onDigitAppended(c.toString())
-        input += c
-        if (formatAsYouType) {
-            digits?.setText(formatter?.inputDigit(c))
+    private fun formatNumber(dialString: String): String {
+        var number = PhoneNumberUtils.extractNetworkPortion(dialString)
+        // Also retrieve the post dial portion of the provided data, so that the entire dial
+        // string can be reconstituted later.
+        val postDial = PhoneNumberUtils.extractPostDialPortion(dialString)
```
Comment (bmarty) [inline vector/src/main/java/im/vector/app/features/call/dialpad/DialPadFragment.kt:180]: Do this after the next `if` block?

File: vector/src/main/java/im/vector/app/features/call/dialpad/DialPadFragment.kt [hunk 2]
```diff
@@ -145,40 +158,40 @@ class DialPadFragment : Fragment() {
...
+        digits.setText("")
     }
 
-    private fun append(c: Char) {
-        callback?.onDigitAppended(c.toString())
-        input += c
-        if (formatAsYouType) {
-            digits?.setText(formatter?.inputDigit(c))
+    private fun formatNumber(dialString: String): String {
+        var number = PhoneNumberUtils.extractNetworkPortion(dialString)
```
Comment (bmarty) [inline vector/src/main/java/im/vector/app/features/call/dialpad/DialPadFragment.kt:177]: `val networkPortion` for clarity?

**Review by bmarty — APPROVED** [review]
Just some few remarks about code clarity

File: vector/src/main/java/im/vector/app/features/call/dialpad/DialPadFragment.kt (hunk 1 above)
Comment (ganfra) [reply vector/src/main/java/im/vector/app/features/call/dialpad/DialPadFragment.kt]: Done

File: vector/src/main/java/im/vector/app/features/call/dialpad/DialPadFragment.kt (hunk 2 above)
Comment (ganfra) [reply vector/src/main/java/im/vector/app/features/call/dialpad/DialPadFragment.kt]: Done

File: vector/src/main/java/im/vector/app/features/call/dialpad/DialPadFragment.kt [hunk 3]
```diff
@@ -145,40 +158,39 @@ class DialPadFragment : Fragment() {
...
-            digits?.setText(formatter?.inputDigit(c))
+    private fun formatNumber(dialString: String): String {
+        val networkPortion = PhoneNumberUtils.extractNetworkPortion(dialString)
+        if (TextUtils.isEmpty(networkPortion)) {
+            return ""
+        }
+        val number = PhoneNumberUtils.formatNumber(networkPortion, null, regionCode) ?: networkPortion
+        // Also retrieve the post dial portion of the provided data, so that the entire dial string can be reconstituted
+        val postDial = PhoneNumberUtils.extractPostDialPortion(dialString)
+        return if (postDial.isEmpty()) {
```
Comment (bmarty) [inline vector/src/main/java/im/vector/app/features/call/dialpad/DialPadFragment.kt:184]: Can't you just return `number + postDial`? Should be ok if `postDial` is empty, no?

Comment (ganfra) [reply vector/src/main/java/im/vector/app/features/call/dialpad/DialPadFragment.kt]: Probably :O

**Pull request merged**